bindings/src/main/java/io/pravega/storage/hdfs/HDFSChunkStorage.java

```diff
@@ -0,0 +1,360 @@
+        return true;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region AutoCloseable Implementation
+
+    @Override
+    public void close() {
+        if (!this.closed.getAndSet(true)) {
+            if (this.fileSystem != null) {
+                try {
+                    this.fileSystem.close();
+                    this.fileSystem = null;
+                } catch (IOException e) {
+                    log.warn("Could not close the HDFS filesystem: {}.", e);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            FileStatus last = fileSystem.getFileStatus(getFilePath(chunkName));
+            ChunkInfo result = ChunkInfo.builder().name(chunkName).length(last.getLen()).build();
+            return result;
+        } catch (IOException e) {
+            throwException(chunkName, "doGetInfo", e);
+        }
+        return null;
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            Path fullPath = getFilePath(chunkName);
+            // Create the file, and then immediately close the returned OutputStream, so that HDFS may properly create the file.
+            this.fileSystem.create(fullPath, READWRITE_PERMISSION, false, 0, this.config.getReplication(),
+                    this.config.getBlockSize(), null).close();
+            log.debug("Created '{}'.", fullPath);
+
+            // return handle
+            return ChunkHandle.writeHandle(chunkName);
+        } catch (FileAlreadyExistsException e) {
+            throw new ChunkAlreadyExistsException(chunkName, "HDFSChunkStorage::doCreate");
+        } catch (IOException e) {
+            throwException(chunkName, "doCreate", e);
+        }
+        return null;
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        FileStatus status = null;
+        try {
+            status = fileSystem.getFileStatus(getFilePath(chunkName));
+        } catch (IOException e) {
+            // HDFS could not find the file. Returning false.
+            log.warn("Got exception checking if file exists", e);
+        }
+        boolean exists = status != null;
+        return exists;
+    }
+
+    @Override
+    protected void doDelete(ChunkHandle handle) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            boolean retValue = this.fileSystem.delete(getFilePath(handle.getChunkName()), true);
```

Comment:
it doesn't look like we need this `boolean retValue` declaration.

Reply:
fixed